src/common/date-helpers.ts

```diff
@@ -0,0 +1,48 @@
+export function getDayNames (firstDay: number = 0, nameLength: string = 'short', locale: string = 'en-US'): Array<string> {
+    // Days start from Sunday (Sunday = 0, Monday = 1, etc.)
+    const listOfDayNames: Array<string> = [];
+
+    // For each of the seven days of the week
+    for (let i = 0; i < 7; i++) {
+        let dayName = new Date(1970, 5, i + firstDay).toLocaleString(locale, { weekday: nameLength });
+        listOfDayNames.push(dayName);
+    }
+
+    return listOfDayNames;
+}
```

Comment:
I think this is quite subjective. Personally, I find the for-loops more readable. I did change some of the variable names so perhaps it's a bit easier to read now.